lti_consumer/signals.py

```diff
@@ -62,3 +62,42 @@ def publish_grade_on_score_update(sender, instance, **kwargs):  # pylint: disabl
                 exc,
             )
             raise exc
+
+
+@receiver(pre_save, sender=LtiConfiguration, dispatch_uid='update_xblock_lti_configuration')
+def update_xblock_lti_configuration(sender, instance, **kwargs):  # pylint: disable=unused-argument
+    """
+    Update the XBlock settings with the changes from the corresponding LtiConfiguration object.
+
+    This provides a way to synchronise the changes done to the LtiConfiguration via admin
+    interface reflect on the XBlock settings in Studio. A complementary hook is implemented in
+    the XBlock to update the LtiConfiguration whenever the XBlock settings are updated.
+    """
+    lti_1p3_values = [
+        "lti_1p3_launch_url",
+        "lti_1p3_oidc_url",
+        "lti_1p3_tool_key_mode",
+        "lti_1p3_tool_keyset_url",
+        "lti_1p3_tool_public_key",
+        "lti_1p3_enable_nrps",
+        "lti_1p3_block_key",
+        "lti_advantage_deep_linking_enabled",
+        "lti_advantage_deep_linking_launch_url",
+        "lti_advantage_ags_mode",
+        "weight",
+        "display_name",
+    ]
+    if not instance.id:
+        # New entry is being created. Do Nothing
+        return
+
+    # We are concerned only with updating the LTI 1.3 values stored in lti_config (for now)
+    # TODO: Should we expand this criteria to include other attributes like version?
+    #
+    previous = LtiConfiguration.objects.get(pk=instance.id)
+    if previous.lti_config != instance.lti_config:
+        block = instance.block
```

Comment:
@tecoholic What if this block is configured in the database only, and `instance.block` is `None`? This will fail saving changes. 
This case is specially important given we want to make LTI 1.3 launches independent from blocks - and we won't have `block` defined for all `LtiConfiguration` instances.

Reply:
@giovannicimolin This Signal is for a way to update the XBlock's settings when the change occurs in the DB. If the block is `None` then we don't have the scenario where we need to keep the XBlock in sync. So performing no action at that point, I think is the valid action.